File: python/non-graphical/prime_factors.py
#!/usr/bin/env python3
from math import sqrt

class prime_factors(object):
    def __init__(self):
        self.factors = []
        self.run()

    def decompose(self, n):
        while(n%2 == 0):
            self.factors.append(2)
            n /= 2

        for i in range(3,int(sqrt(n))+1):
            while(n%i == 0):
                self.factors.append(i)
                n /= i

        if(n > 2):
            self.factors.append(int(n))

    def run(self):
        with open("rand_ints.txt", mode='r') as READ_FILE:
            for line in READ_FILE:
                self.decompose(int(line.strip()))
        READ_FILE.close()


def main():
	pass

if __name__=='__main__':
    main()

File: cpp/non-graphical/prime_factors.cpp
#include <iostream>
#include <math.h>
#include <vector>
#include <string>
#include <fstream>
#include "prime_factors.h"
using namespace std;

prime_factors::prime_factors(){
  run();
}

void prime_factors::decompose(long n){
  while(n%2 == 0){
      factors.push_back(2);
      n = n/2;
  }

  for(int i=3; i<sqrt(n); i++){
      while(n%i == 0){
          factors.push_back(i);
          n = n/i;
      }
  }

  if(n>2){
      factors.push_back(n);
  }
}

void prime_factors::run(){
  std::ifstream in_file;
  std::string line;
  in_file.open("rand_ints.txt");
  if(!in_file.is_open()){
    cout << "unable to open file" << endl;
    return;
  }
  while(getline(in_file, line)){
    long n = stol(line);
    decompose(n);
  }
  in_file.close();
}
